CRM/Contribute/Form/AdditionalPayment.php

```diff
@@ -392,79 +398,29 @@ public function submit($submittedValues) {
     CRM_Core_Session::setStatus($statusMsg, ts('Saved'), 'success');
   }
 
-  /**
-   * @param $submittedValues
-   */
-  public function processCreditCard($submittedValues) {
+  public function processCreditCard() {
     $config = CRM_Core_Config::singleton();
     $session = CRM_Core_Session::singleton();
 
-    $unsetParams = array(
-      'trxn_id',
-      'payment_instrument_id',
-      'contribution_status_id',
-    );
-    foreach ($unsetParams as $key) {
-      if (isset($submittedValues[$key])) {
-        unset($submittedValues[$key]);
-      }
-    }
-
-    // Get the required fields value only.
-    $params = $submittedValues;
-
     $now = date('YmdHis');
     $fields = array();
 
     // we need to retrieve email address
-    if ($this->_context == 'standalone' && !empty($submittedValues['is_email_receipt'])) {
+    if ($this->_context == 'standalone' && !empty($this->_params['is_email_receipt'])) {
       list($this->userDisplayName,
         $this->userEmail
         ) = CRM_Contact_BAO_Contact_Location::getEmailDetails($this->_contactId);
       $this->assign('displayName', $this->userDisplayName);
     }
 
-    //set email for primary location.
-    $fields['email-Primary'] = 1;
-    $params['email-Primary'] = $this->_contributorEmail;
-
-    // now set the values for the billing location.
-    foreach ($this->_fields as $name => $dontCare) {
-      $fields[$name] = 1;
-    }
-
-    // also add location name to the array
-    $params["address_name-{$this->_bltID}"] = CRM_Utils_Array::value('billing_first_name', $params) . ' ' . CRM_Utils_Array::value('billing_middle_name', $params) . ' ' . CRM_Utils_Array::value('billing_last_name', $params);
-    $params["address_name-{$this->_bltID}"] = trim($params["address_name-{$this->_bltID}"]);
-    $fields["address_name-{$this->_bltID}"] = 1;
-
-    $ctype = civicrm_api3('Contact', 'getvalue', array(
-      'return' => "contact_type",
-      'id' => $this->_contactId,
-    ));
-
-    $nameFields = array('first_name', 'middle_name', 'last_name');
-    foreach ($nameFields as $name) {
-      $fields[$name] = 1;
-      if (array_key_exists("billing_$name", $params)) {
-        $params[$name] = $params["billing_{$name}"];
-        $params['preserveDBName'] = TRUE;
-      }
-    }
-
-    if (!empty($params['source'])) {
-      unset($params['source']);
-    }
-    $contactID = CRM_Contact_BAO_Contact::createProfileContact($params, $fields,
-      $this->_contactId,
-      NULL, NULL,
-      $ctype
+    $fields = $this->formatParamsForPaymentProcessor($this->_params);
```

Comment:
ok - so using the parent function to create the profile contact outside the credit card processing didn't work?